File: C++/Programmers/타겟 넘버.cpp
#include <string>
#include <vector>

using namespace std;

int answer = 0;

void dfs(vector<int> numbers, int sum, int cnt, int target){
    if(cnt == numbers.size()){
        if(sum == target){
            answer++;
            return;
        } else {
            return;
        }
    }
    
    dfs(numbers, sum + numbers[cnt], cnt+1, target);
    dfs(numbers, sum - numbers[cnt], cnt+1, target);
}

int solution(vector<int> numbers, int target) {
    dfs(numbers, 0,0,target);
    return answer;
}

File: Python/BFSDFS/타겟넘버.py
answer = 0

def dfs(numbers, sum, cnt, target):
    global answer
    if cnt == len(numbers):
        if sum == target:
            answer += 1
            return
        else:
            return
    
    dfs(numbers, sum + numbers[cnt], cnt+1, target)
    dfs(numbers, sum - numbers[cnt], cnt+1, target)
    

def solution(numbers, target):
    dfs(numbers, 0, 0, target)
    return answer
